src/intervals/intervals.jl

```diff
@@ -106,9 +102,10 @@ end
 
 `interval(a, b)` checks whether [a, b] is a valid `Interval`, which is the case if `-∞ <= a <= b <= ∞`, using the (non-exported) `is_valid_interval` function. If so, then an `Interval(a, b)` object is returned; if not, then an error is thrown.
 """
-function interval(a::Real, b::Real)
+function interval(a::T, b::S) where {T<:Real, S<:Real}
     if !is_valid_interval(a, b)
-        throw(ArgumentError("`[$a, $b]` is not a valid interval. Need `a ≤ b` to construct `interval(a, b)`."))
+        @warn "Invalid input, empty interval is returned"
+        return emptyinterval(promote_type(T, S, Float64))
```

Comment:
The idea of having `interval` was to check the validity of the construction, which does not happen if you use `Interval`. This yields some speed-ups since the `is`s are avoided. While I think it is ok to return a `nai` for `DecoratedInterval`s, I think that we should return an error for bare intervals.

Reply:
`Interval` still does not make any check, so e.g. `Interval(3, 1)` works like before. The validity check is only in `interval` (and in the methods that depend on it) so for example
before:
```
 julia> Interval(3, 1)
[3, 1]

julia> interval(3, 1)
ERROR: ArgumentError: `[3, 1]` is not a valid interval. Need `a ≤ b` to construct `interval(a, b)`.
Stacktrace:
 [1] interval(a::Int64, b::Int64)
   @ IntervalArithmetic ~\.julia\dev\IntervalArithmetic\src\intervals\intervals.jl:111
 [2] top-level scope
   @ REPL[60]:1

```

after:
```
julia> Interval(3, 1)
[3, 1]

julia> interval(3, 1)
┌ Warning: Invalid input, empty interval is returned
└ @ IntervalArithmetic C:\Users\lucaa\.julia\dev\IntervalArithmetic\src\intervals\intervals.jl:107
∅
```

Reply:
That's precisely my point: I think using `interval` should throw an error, instead of a warning, as it is proposed. This behavior could be changed for a specific "flavor".

Reply:
The motivation for returning an empty interval was the standard extract posted in https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/457#issuecomment-830855196

My understanding from that was that it should return an empty interval and signal the behavior (print the warning). Mathematically the set `{x in R | 2<=x<=1}` is the empty set, so I see no reason why `2..1` should not return an empty interval. 

I guess throwing an error might be more useful for debugging though. If somewhere in the code you try to build an invalid interval, interrupting the computation and pointing to the line where this happened would be more useful than silently constructing an empty interval and let it propagate through all computations.